substrate/runtime-support/src/dispatch.rs

```diff
@@ -611,13 +370,13 @@ macro_rules! __impl_outer_dispatch_common {
 #[doc(hidden)]
 macro_rules! __impl_json_metadata {
 	(
-		impl for $mod_type:ident<$trait_instance:ident: $trait_name:ident>;
+		$mod_type:ident $trait_instance:ident $trait_name:ident
 		$($rest:tt)*
 	) => {
 		impl<$trait_instance: $trait_name> $mod_type<$trait_instance> {
 			pub fn json_metadata() -> &'static str {
-				concat!(r#"{ "name": ""#, stringify!($mod_type), r#"", "calls": ["#,
-					__calls_to_json!(""; $($rest)*), " ] }")
+				concat!(r#"{ "name": ""#, stringify!($mod_type), r#"", "calls": [ "#,
```

Comment:
If we do not have multiple call I would change the json from `"calls": [ { .. }]` to `"call": { .. }`.